app/src/main/java/org/wikipedia/diff/ArticleEditDetailsFragment.kt

```diff
@@ -463,16 +442,72 @@ class ArticleEditDetailsFragment : Fragment(), WatchlistExpiryDialog.Callback, L
+                            (item.diff.lineNumber - lastItem!!.diff.lineNumber == 1 && lastItem!!.diff.type == DiffResponse.DIFF_TYPE_LINE_WITH_SAME_CONTENT && item.diff.type == DiffResponse.DIFF_TYPE_LINE_WITH_SAME_CONTENT) ||
+                            (lastItem!!.diff.type == DiffResponse.DIFF_TYPE_LINE_REMOVED && item.diff.type == DiffResponse.DIFF_TYPE_LINE_REMOVED))) {
+                val str = SpannableStringBuilder(lastItem!!.parsedText)
+                str.append("\n")
+                str.append(item.parsedText)
+                lastItem!!.parsedText = str
+            } else {
+                items.add(item)
+            }
+            lastItem = item
+        }
+        // val items = diffList.map { DiffLine(it) }.filter { it.parsedText.toString().trim().isNotEmpty() || it.diff.type != DiffResponse.DIFF_TYPE_LINE_WITH_SAME_CONTENT }
+        binding.diffRecyclerView.adapter = DiffLinesAdapter(items)
+    }
+
+    inner class DiffLine(diff: DiffResponse.DiffItem) {
+        val diff: DiffResponse.DiffItem
+        var parsedText: CharSequence
+        var expanded: Boolean
+
+        init {
+            parsedText = createSpannable(diff)
+            this.diff = diff
+            expanded = diff.type != DiffResponse.DIFF_TYPE_LINE_WITH_SAME_CONTENT
+        }
+    }
 
-        fun newInstance(articleTitle: String, revisionId: Long, languageCode: String): ArticleEditDetailsFragment {
+    private inner class DiffLinesAdapter(val diffLines: List<DiffLine>) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+        override fun getItemCount(): Int {
+            return diffLines.size
+        }
+
+        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+            return DiffLineHolder(DiffLineView(requireContext()))
+        }
+
+        override fun onBindViewHolder(holder: RecyclerView.ViewHolder, pos: Int) {
+            if (holder is DiffLineHolder) {
+                holder.bindItem(diffLines[pos])
+            }
+            holder.itemView.tag = pos
+        }
+    }
+
+    private inner class DiffLineHolder constructor(itemView: DiffLineView) : RecyclerView.ViewHolder(itemView) {
+        fun bindItem(item: DiffLine) {
+            (itemView as DiffLineView).setItem(item)
+        }
+    }
+
+    companion object {
+        fun newInstance(articleTitle: String, revisionFrom: Long, revisionTo: Long,
+                        languageCode: String): ArticleEditDetailsFragment {
             return ArticleEditDetailsFragment().apply {
-                arguments = bundleOf(EXTRA_ARTICLE_TITLE to articleTitle,
-                        EXTRA_EDIT_REVISION_ID to revisionId,
-                        EXTRA_EDIT_LANGUAGE_CODE to languageCode)
+                arguments = bundleOf(ArticleEditDetailsActivity.EXTRA_ARTICLE_TITLE to articleTitle,
```

Comment:
Found a language variant issue when trying to fix this one: #3183, would it be possible to pass `PageTitle` instead of plain `String` here? And maybe we can reduce some redundant work of sending `languageCode` or creating `WikiSite` if we send `PageTitle` here.

Reply:
correct!